### PR TITLE
Modernize build toolchain with Rollup

### PR DESCRIPTION
**Pull request by maxburs**
Update build toolchain while keeping built artifacts and dev process mostly the same.

## Changes
* Rollup replaces `requirejs` and `monaco-plugin-helpers` in our build script
  * `kustoWorker.js` and `kustoMode.js` now share code in a `main` file, instead of duplicating code (1)
* When not in CI, source maps are now created
* `yarn watch` works again
  * Uses Rollup to build .js files instead of `tsc`
* /release/esm no longer bundles /node_modules 

(1)

### CODE REVIEW

File: text.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+posted in Kusto Web Explorer / General at February 9, 2023 6:52:54 PM PST In case you missed it 👆 What's the email of copy editor?
```
Comment (noam-shapira) [inline text.txt:1]: care to explain ?

File: package/tsconfig.json [hunk 2]
```diff
@@ -3,15 +3,16 @@
     "include": ["./src/**/*"],
     "compilerOptions": {
         "rootDir": "./src",
-        "module": "amd",
+        "module": "esnext",
```
Comment (noam-shapira) [inline package/tsconfig.json:6]: es5 please

**Review by noam-shapira — APPROVED** [review]
LGTM

**Comment by noam-shapira** [comment]
Another question how come the kustoMode size is reduced from 118K to 39K? is it tree-shaking?

File: text.txt (hunk 1 above)
Comment (maxburs) [reply text.txt:1]: Woops, didn't mean to commit this. I'll delete it.

File: package/scripts/rollup.dev.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    return {
+        external: ['monaco-editor-core'],
+        input: Object.fromEntries(entryPointsAMD.map((e) => [e, path.join(packageFolder, 'src', e + '.ts')])),
+        plugins: [
+            nodeResolve({ extensions }),
+            commonJs(),
+            babel({
+                extensions,
+                babelHelpers: 'inline',
+                presets: [['@babel/preset-env', { targets: { ie: 11 } }], '@babel/preset-typescript'],
```
Comment (maxburs) [inline package/scripts/rollup.dev.js:26]: Note: Add a comment that @babel/preset-env transplies for ebiza

**Comment by maxburs** [comment]
> Another question how come the kustoMode size is reduced from 118K to 39K? is it tree-shaking?

I've updated imports with `type` when possible, which caused some code to not be included in that bundle.

File: package/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "@kusto/monaco-kusto",
-    "version": "5.6.6",
+    "version": "5.7.0",
```
Comment (izikl) [inline package/package.json:3]: if it breaks something, let's change it to 6.x

File: package/package.json [hunk 5]
```diff
@@ -18,49 +18,59 @@
     },
     "scripts": {
         "build": "ts-node ./scripts/build.ts",
-        "watch": "echo dev > test/mode.txt && tsc -p ./tsconfig.watch.json && npm run copy_runtime_deps_to_out && concurrently \"live-server ./ \" \"tsc -w -p ./tsconfig.watch.json \"",
+        "watch": "ts-node ./scripts/watch.ts",
+        "typecheck": "tsc",
         "test_release": "echo release > test/mode.txt && http-server -c-1 -p 8080 ./ -o index.html",
         "test": "jest",
-        "clean": "mrmdir ./out && mrmdir ./release"
+        "clean": "rimraf ./out ./release"
```
Comment (izikl) [inline package/package.json:25]: the readme mention yarn start , but there is no start script.

File: package/tsconfig.json (hunk 2 above)
Comment (maxburs) [reply package/tsconfig.json:6]: This is just used for type-checking now. It's still getting transpiled, the config now lives in /package/scripts/rollup.amd.js

**Review by izikl — APPROVED** [review]

**Pull request merged**